tools/lexical-model.ts

```diff
@@ -0,0 +1,35 @@
+
+interface LexicalModel {
+  readonly format: 'trie-1.0'|'fst-foma-1.0'|'custom-1.0',
+  readonly wordBreaking?: {
+    allowedCharacters?: { initials?: string, medials?: string, finals?: string } | string,
+    defaultBreakCharacter?: string
+    sources?: Array<string>;
+  },
+  //... metadata ...
+}
+
+interface LexicalModelPrediction {
+  display?: string;
+  transform: string;
+  delete: number;
+}
+
+interface LexicalModelSource extends LexicalModel {
+  readonly sources: Array<string>;
+}
+
+interface LexicalModelCompiled extends LexicalModel {
+  readonly id: string;
+}
+
+interface LexicalModelCompiledTrie extends LexicalModelCompiled {
+  trie: string;
+}
+
+interface LexicalModelCompiledFst extends LexicalModelCompiled {
+  fst: string;
+}
+
+interface LexicalModelCompiledCustom extends LexicalModelCompiled {
```

Comment:
This will most likely also contain `src: string`, but we'll cross that bridge when we get there :D